src/util/threadpool.rs

```diff
@@ -295,9 +319,11 @@ impl<Q, T> Worker<Q, T>
         let mut task = self.get_next_task(None);
         // Start the worker. Loop breaks when receive stop message.
         while let Some(t) = task {
+            t.ctx.on_start();
             // Since tikv would be down when any panic happens,
             // we don't need to process panic case here.
-            (t.task)();
+            (t.task)(t.ctx.clone());
```

Comment:
It's very embarrassed, there is a rust compiler bug about call a Box<FnBox<_>> with a reference argument. https://github.com/rust-lang/rust/issues/25647
Current implementation also has a bug. I think I should find another way to pass the context.
@BusyJay